platforms/cpuinfo_linux_arm_test.go

```diff
@@ -0,0 +1,33 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package platforms
+
+func TestGetCPUArchitecture(t *testing.T) {
+	p, err := getCPUArchitecture()
+	if err != nil {
+		t.Fatal(err)
+	}
+	variants := []string{"v8", "v7", "v6", "v5", "v4", "v3"}
```

Comment:
I'm wondering how useful this test is in its current form; 

- CI definitely can't test on all possible variants (as we don't have hardware for this)
- if `getCPUArchitecture()` would return anything else than the "v8", "v7", "v6", "v5", "v4", "v3" variants (e.g. if it would return `7m`, it would still hit the normalisation logic [at the end of `getCPUVariant()`](https://github.com/containerd/containerd/blob/77d53d2d230c3bcd3f02e6f493019a72905c875b/platforms/cpuinfo.go#L113-L119)

So _either_;

- values like `7m` or `7` (without `v` prefix) should be considered "valid"
- or `getCPUArchitecture()` should include the normalisation logic
- or (if it can only return _valid_ values), we should make ` getCPUVariant()` return early, and return the value from `getCPUArchitecture()` as-is (if there's no error